drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -2611,6 +2611,11 @@ Error RenderingDeviceDriverVulkan::command_queue_execute_and_present(CommandQueu
 		// it'll lead to very low performance in Android by entering an endless loop where it'll always resize the swap chain
 		// every frame.
 
+		ERR_FAIL_COND_V(err == VK_ERROR_OUT_OF_HOST_MEMORY, FAILED);
+		ERR_FAIL_COND_V(err == VK_ERROR_OUT_OF_DEVICE_MEMORY, FAILED);
+		ERR_FAIL_COND_V(err == VK_ERROR_DEVICE_LOST, FAILED);
+		ERR_FAIL_COND_V(err == VK_ERROR_SURFACE_LOST_KHR, FAILED);
+		ERR_FAIL_COND_V(err == VK_ERROR_FULL_SCREEN_EXCLUSIVE_MODE_LOST_EXT, FAILED);
 		ERR_FAIL_COND_V(err != VK_SUCCESS && err != VK_SUBOPTIMAL_KHR, FAILED);
```

Comment:
Instead of checking every condition it would be best to do the following:
1. print a string using `ERR_FAIL_COND_V_MSG()`
2. Create a function that takes in a `VkResult` and returns a string. The implementation of the function can be hidden behind an `#ifdef DEBUG` so that the full error handling is only done for debug builds. we use this pattern in a few places in the source (here for example: https://github.com/godotengine/godot/blob/87318a2fb7fffeb72adca934e31915be077c3d1f/drivers/gles3/storage/texture_storage.h#L706)